### ISSUE TITLE
Не работает Looper на stm32 arduino 

### ISSUE THREAD

**Issue by JonyBest** [GyverLibs/Looper#1]
Доброго времени! На esp32 библиотека работает великолепно. Настолько хороша, что вызывает привыкание :)
Но проекты на stm32f401cc, Arduino Core for STM32 2.8.1, не работают.
```
LP_TIMER(500, [](){
    Serial.println("hello!");
});
```
Таймер не пускается. В порт print не выводит
```
LP_THREAD({
if (Looper.thisSetup()) {
}
Serial.println("myPins");
 })
```
Этот код приводит к зависанию

**Comment by GyverLibs** [comment]
Привет, а обычный task работает?

**Comment by JonyBest** [comment]
Да. Пример [https://github.com/GyverLibs/GyverOS/blob/main/examples/demo](url) работает

**Comment by GyverLibs** [comment]
Так это gyveros

**Comment by JonyBest** [comment]
На библиотеке Looper 
```
LP_TICKER_("tick0", []() {
    switch (Looper.thisState()) {
        case tState::Setup:
            Serial.println("SETUP");
            break;

        case tState::Loop:
            Serial.println("LOOP");
            Looper.thisTask()->removeLoop();
            break;

        case tState::Exit:
            Serial.println("EXIT");
            break;

        case tState::Event:
            Serial.println("EVENT");
            break;

        default: break;
    }
});
```
этот пример не работает, мк виснет
Я не пойму поставленной задачи. Покажите нужный пример - я его протестирую.

**Comment by GyverLibs** [comment]
Какой задачи?

**Comment by Gemerus** [comment]
У меня тоже не работает, никакой пример не запускается.

**Comment by GyverLibs** [comment]
Отладьте в сериал, на какой строке в либе зависает

**Comment by GyverLibs** [comment]
или дайте инфу как завести ардуино на блюпилл, в каком софте на каком sdk. Насколько я помню там несколько пакетов поддержки, все разные, загружается и работает всё по разному, какие то дрова надо искать

**Comment by Gemerus** [comment]
В общем нельзя как в примере прост написать LP_THREAD.... где то перед Setup, это не работает. Но если все LP_THREAD и тд положить в функцию и вызвать её в конце Setup то всё отлично работает. Platformio,STM32 arduino core

**Comment by GyverLibs** [comment]
хм, интересно. В общем попробую сам завести блюпилку на платформио

**Comment by GyverLibs** [comment]
есть какой то вариант отладки при подключении через STlink v2 по SWD? Или по бортовому USB хотя бы?

**Comment by JonyBest** [comment]
Вариант отладки проще организовать через Serial.print используя для вывода STlink v2,1( у него есть RX, TX) или STlink v2 для программирования STM32 + любой адаптер USB в TTL/UART. По умолчанию RX - PA3, TX - PA2.

**Comment by GyverLibs** [comment]
Порнуха) попробую
